packages/aws-cdk/lib/toolkit/cli-io-host.ts

```diff
@@ -21,7 +21,21 @@ interface IoMessage {
   readonly action: IoAction;
 
   /**
-   * A short code uniquely identifying message type.
+   * A short message code uniquely identifying a message type of the format [CATEGORY]_[NUMBER_CODE].
```

Comment:
I think this is fine but I like what @rix0rrr [suggested](https://github.com/aws/aws-cdk/pull/32708/files#r1905163106) with using letters to denote level.

So it would be: `[CATEGORY]_[LEVEL_CHAR][NUMBER_CODE]`

- `ASSETS_E0001`
- `TOOLKIT_I0001`

Reply:
Did you have a look at how JSII [emits messages](https://github.com/aws/jsii-compiler/blob/main/src/jsii-diagnostic.ts)? Not saying we should do exactly the same - I think this proposal actually adds some enhancements over it, but its a good resource to look at.

Reply:
I didn't really check it out and just started fresh. But if we want to incorporate that into the number system that's fine with me